Client/Frontend/Browser/Tab.swift

```diff
@@ -679,3 +682,66 @@ class TabWebViewMenuHelper: UIView {
         }
     }
 }
+
+// MARK: - Brave Search
+
+extension Tab {
+    /// Call the api on the Brave Search website and passes the fallback results to it.
+    /// Important: This method is also called when there is no fallback results
+    /// or when the fallback call should not happen at all.
+    /// The website expects the iOS device to always call this method(blocks on it).
+    func injectResults() {
+        let dispatchGroup = DispatchGroup()
+        var shouldInjectResults = true
+        
+        dispatchGroup.enter()
+        DispatchQueue.main.async {
+            // If the backup search results happen before the Brave Search loads
+            // The method we pass data to is undefined.
+            // For such case we do not call that method or remove the search backup manager.
+            // swiftlint:disable:next safe_javascript
+            self.webView?.evaluateJavaScript("window.onFetchedBackupResults === undefined") {
```

Comment:
note that any website can pass this check by simply defining `window.onFetchedBackupResults`.